cmudb/tscout/clang_parser.py

```diff
@@ -266,8 +273,36 @@ def _construct_fully_expanded_fields(self, class_name, classes, prefix=""):
                     expanded_fields = self._construct_fully_expanded_fields(
                         field.pg_type, classes, prefix=prefix + f"{field.name}_"
                     )
+
                     new_fields.extend(expanded_fields)
+
+                    # This is a big band-aid, aimed at temporarily patching the behavior of hutch/TScout when
+                    # it comes to struct alignments. As of this change (Mar 7, 2022), we have that BPF ->
+                    # Python userspace structs don't respect __attribute__((aligned(#))). It doesn't seem like
+                    # ctypes has a way of specifying per-field alignment either...
+                    #
+                    # We can theoretically generate structs for the python collector to read but then Hutch would
+                    # still segfault with the incorrect offsets. The following attempts to compute whether or not
+                    # there will be padding between the last data variable and the end of the struct.
+                    #
+                    # If there is a difference, a manual "padding" field is inserted.
+                    struct_size = classes[field.pg_type].type.get_size()
+                    last_field = rtti_map[field.pg_type][-1]
+                    last_field_offset = int(classes[field.pg_type].type.get_offset(last_field.name) / 8)
+                    last_field_data = last_field.field_size + last_field_offset
+                    if struct_size - last_field_data > 0:
+                        padding = struct_size - last_field_data
+                        new_fields.append(
+                            Field(
+                                name=f"{prefix}{field.name}_padding",
```

Comment:
Just confirming: there shouldn't be any need to add a number suffix since we should only have 1 padding field per-struct, right?

Reply:
Yes. There should only be at-most 1 padding field per-struct.